src/resolve.jl

```diff
@@ -365,14 +365,21 @@ function offline()
     end
 end
 
+function is_clean(conda_env, meta)
+    meta === nothing && return false
+    conda_env == meta.conda_env || return false
+    stat(conda_env).mtime ≤ meta.timestamp || return false
+    isdir(conda_env) && return true
+    (isempty(meta.packages) && isempty(meta.pip_packages)) && return true
+    false
+end
+
 function resolve(; force::Bool=false, io::IO=stderr, interactive::Bool=false, dry_run::Bool=false)
     # if frozen, do nothing
-    if STATE.frozen
-        return
-    end
+    STATE.frozen && return
     # if backend is Null, assume resolved
     back = backend()
-    if back == :Null
+    if back === :Null
```

Comment:
Using `==` instead of `===` can cause some invalidations (e.g. https://github.com/JuliaPlots/Plots.jl/pull/4622). Furthermore, it is a usual convention to use triples for symbol comparisons.

Reply:
TIL, it even says to use `===` in the Julia docs.